### PR TITLE
roe-413 Update Entity in web model

### PR DESCRIPTION
**Pull request by mattch1**
### JIRA link

https://companieshouse.atlassian.net/browse/ROE-413

### Change description

Added field: is_on_register_in_country_formed_in

Removed address fields from model and re-reference common address models

Ensure that all fields in the model and template match the api spec

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: views/entity.html [hunk 1]
```diff
@@ -96,13 +97,11 @@ <h1 class="govuk-heading-xl">Tell us about the overseas entity</h1>
             text: "Yes",
             conditional: {
                html: entityPublicRegisterHtml
-            },
-            checked: is_on_sanctions_list === 1
+            }
```
Comment (mouhajer-ch) [inline views/entity.html:100]: Hi Matthew, you should not remove the `checked` field, but you should update the controller. You can find example in other fields

**Review by mouhajer-ch — COMMENTED** [review]
I would advice you to update first the SDK with the new field on Entity model and after do all the WEB changes

File: views/entity.html (hunk 1 above)
Comment (mattch1) [reply views/entity.html]: Put in a pr for the sdk: https://github.com/companieshouse/api-sdk-node/pull/384

Added checked fields to two templates as this was based on bo other and that was also missing them controller updated as well.

**Review by mouhajer-ch — COMMENTED** [review]
Thanks for the update, may I ask you please to add an include for `is_on_register_in_country_formed_in` field and remove the condition with the phoneHtml that is not necessary. It is used on BO other and Entity page. Because the text will change add as you know the` {% set registerInCountryFormedText = "Is the other legal entity already on a public register in the country it was formed in?" %}`

Should be something like this

```

        {{ govukRadios({
          classes: "govuk-radios",
          idPrefix: "is_on_register_in_country_formed_in",
          name: "is_on_register_in_country_formed_in",
          fieldset: {
            legend: {
              text: registerInCountryFormedText,
              isPageHeading: false,
              classes: "govuk-fieldset__legend--m"
            }
          },
          items: [
            {
              value: 1,
              text: "Yes",
              conditional: {
                html: entityPublicRegisterHtml
              },
              checked: is_on_register_in_country_formed_in === 1
            },
            {
              value: 0,
              text: "No",
              checked: is_on_register_in_country_formed_in === 0
            }
          ]
        }) }}
```

**Comment by mattch1** [comment]
I have had a look at the includes and what I would like to do is raise a separate ticket.

The reason for this is I think it would be best to include all the registration fields in this includes file together

public_register_name
registration_number
and is_on_register_in_country_formed_in

However, in order to do this we'd need some consistencey in bo other the name "register_name" is used and in the entity it is "public_register_name". A discussion on this came up yesterday in 

https://github.com/companieshouse/overseas-entities-api/pull/44

We are thinking of adding a ticket to do this work and we should probably keep it together. I can raise a ticket if you want. I'll discuss this with @markpit and @mwestacott as well.

**Comment by mattch1** [comment]
What I can do for now is as suggested have an includes file just for the radio buttons but with a view to add those other two feilds as part of this other change.

**Review by mouhajer-ch — APPROVED** [review]

**Pull request merged**